Registry/RegistryParse.py

```diff
@@ -273,6 +290,38 @@ def last_hbin_offset(self):
         """
         return self.unpack_dword(0x28)
 
+    def calculate_checksum(self, include_checksum = False):
+        """
+        checksum is calculated over the first 0x200 bytes
+        XOR of all D-Words from 0x00000000 to 0x000001FB
+        If include_checksum is True, zero is the expected return value for an valid hive.
```

Comment:
took me a minute to figure out how this worked, but now i get it. thats a neat little trick!

i think the optional parameter makes this a little confusing. i'd recommend having a required parameter that is the length of data to checksum, and create a wrapper method that supplies `0x1fb` and has a logical name (`checksum2` perhaps).

using `0x1fb` will allow the `validate` method to be more explicit in checking the checksums (`r.checksum1() == r.checksum2()`, which is probably a good thing for outside readers). otherwise, we don't really even need a method to fetch `checksum1`, which leaves a hole in the data we parse. however, i think you could make an argument for which length to pick.